projects/plugins/jetpack/extensions/shared/external-media/media-button/media-sources.js

```diff
@@ -4,8 +4,19 @@ import {
 	internalMediaSources,
 	externalMediaSources,
 	featuredImageExclusiveMediaSources,
+	generalPurposeImageExclusiveMediaSources,
 } from '../sources';
 
+/**
+ * Temporary feature flag to control generalPurposeImageExclusiveMediaSources
+ * visibility.
+ */
+const GENERAL_PURPOSE_IMAGE_GENERATOR_BETA_FLAG = 'ai-general-purpose-image-generator';
+const isGeneralPurposeImageGeneratorBetaEnabled =
+	window?.Jetpack_Editor_Initial_State?.available_blocks?.[
+		GENERAL_PURPOSE_IMAGE_GENERATOR_BETA_FLAG
+	]?.available === true;
```

Comment:
You can use [this lib function](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/utils/get-feature-availability.ts) here